app/views/products/_product.html.erb

```diff
@@ -0,0 +1,6 @@
+<tr>
+  <td><%= product.name %></td>
+  <td><%= link_to 'Show', product %></td>
+  <td><%= link_to 'Edit', edit_product_path(product) %></td>
+  <td><%= link_to 'Destroy', product, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+</tr>
```

Comment:
Setup vscode tự tạo endline khi save nha e. Rồi check lại các file bị warning endline, ví dụ như file này
https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files